### PR TITLE
Fix: Truncate overflowing dashboard apps text

### LINKED ISSUE #5213: UI  - Dashboard Apps v2.7
When a new "Dashboard" is created, it cannot be edited, deleted and in the web interface it appears like this.

Version 2.7
self-hosted

Screen Shot 2022-08-06 at 17 36 26

### PR DESCRIPTION
**Pull request by fayazara**
# Pull Request Template

## Description
Screenshot2022-08-08 at 15 58 27@2x

Truncates overflowing text in dashboard apps titles and urls.

Fixes #5213 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue [hunk 1]
```diff
@@ -43,3 +49,15 @@ export default {
   },
 };
 </script>
+
+<style lang="scss" scoped>
+.dashboard_app-url {
+  max-width: calc(var(--space-giga) + var(--space-mega));
+}
+.dashboard_app-title {
+  max-width: calc(var(--space-giga) / 2);
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue:58]:
```suggestion
  max-width: calc(var(--space-giga) / var(--space-two));
```

File: app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue [hunk 2]
```diff
@@ -43,3 +49,15 @@ export default {
   },
 };
 </script>
+
+<style lang="scss" scoped>
+.dashboard_app-url {
+  max-width: calc(var(--space-giga) + var(--space-mega));
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue:55]: @fayazara This doesn't seem to be the right fix. What would happen on smaller/larger screens?

Comment (fayazara) [reply app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue:55]: @pranavrajs Another option would be to add `table-layout: auto` and `width:100%`, so words can overflow. 

Which would look something like this, 
Screenshot2022-08-09 at 15 57 44@2x

Comment (pranavrajs) [reply app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue:55]: Hiding content on smaller screen works.

**Comment by fayazara** [comment]
@pranavrajs This seems like a better fix, please have a look.
Screenshot2022-08-11 at 09 59 55

File: app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue [hunk 3]
```diff
@@ -1,7 +1,13 @@
 <template>
   <tr>
-    <td>{{ app.title }}</td>
-    <td>{{ app.content[0].url }}</td>
+    <td>
+      {{ app.title }}
+    </td>
+    <td class="dashboard-app-label-url">
+      <span>
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue:7]:
```suggestion
      <span class="text-truncate">
```

File: app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue [hunk 4]
```diff
@@ -43,3 +49,22 @@ export default {
...
+    content: '&nbsp;';
+    visibility: hidden;
+  }
+  span {
+    position: absolute;
+    left: 0;
+    right: 0;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppsRow.vue:67]:
```suggestion

```

**Review by iamsivin — COMMENTED** [review]
Please see the comments.

**Review by iamsivin — APPROVED** [review]
LGTM

**Review by iamsivin — APPROVED** [review]

**Pull request merged**